contrib/completions/bash/buildah

```diff
@@ -0,0 +1,461 @@
+    local parts=($1)
+    local IFS='|'
+    echo "${parts[*]}"
+}
+
+# Transforms a multiline list of options into an extglob pattern
+# suitable for use in case statements.
+__buildah_to_extglob() {
+    local extglob=$(__buildah_to_alternatives "$1")
+    echo "@($extglob)"
+}
+
+# Subcommand processing.
+# Locates the first occurrence of any of the subcommands contained in the
+# first argument. In case of a match, calls the corresponding completion
+# function and returns 0.
+# If no match is found, 1 is returned. The calling function can then
+# continue processing its completion.
+#
+# TODO if the preceding command has options that accept arguments and an
+# argument is equal ot one of the subcommands, this is falsely detected as
+# a match.
+__buildah_subcommands() {
+	  local subcommands="$1"
+
+	  local counter=$(($command_pos + 1))
+	  while [ $counter -lt $cword ]; do
+		    case "${words[$counter]}" in
+			      $(__buildah_to_extglob "$subcommands") )
+subcommand_pos=$counter
+local subcommand=${words[$counter]}
+local completions_func=_buildah_${command}_${subcommand}
+declare -F $completions_func >/dev/null && $completions_func
+return 0
+;;
+esac
+(( counter++ ))
+done
+return 1
+}
+
+ # suppress trailing whitespace
+ __buildah_nospace() {
+     # compopt is not available in ancient bash versions
+     type compopt &>/dev/null && compopt -o nospace
+ }
+
+
+ # global options that may appear after the buildah command
+ _buildah_buildah() {
+     local boolean_options="
+         $global_boolean_options
+         --help
+         --version -v
+         --debug
+         "
+     local options_with_args="
+         --root
+         --runroot
+         --storage-driver
+         --storage-option
+         "
+
+     case "$prev" in
+         --root | --runroot)
+             case "$cur" in
+                 *:*) ;; # TODO somehow do _filedir for stuff inside the image, if it's already specified (which is also somewhat difficult to determine)
+                 '')
+                     COMPREPLY=($(compgen -W '/' -- "$cur"))
+                     __buildah_nospace
+                     ;;
+                 *)
+                     _filedir
+                     __buildah_nospace
+                     ;;
+             esac
+             return
+             ;;
+
+         --storage-driver)
+             COMPREPLY=($(compgen -W 'devicemapper overlay overlay2' -- "$cur"))
+             return
+             ;;
+
+         $(__buildah_to_extglob "$options_with_args"))
+ return
+ ;;
+ esac
+
+ case "$cur" in
+     -*)
+         COMPREPLY=($(compgen -W "$boolean_options $options_with_args" -- "$cur"))
+         ;;
+     *)
+         local counter=$(__buildah_pos_first_nonflag $(__buildah_to_extglob "$options_with_args"))
+         if [ $cword -eq $counter ]; then
+             COMPREPLY=($(compgen -W "${commands[*]} help" -- "$cur"))
+         fi
+         ;;
+ esac
+}
+
+ _buildah_delete() {
+     local boolean_options="
+     --help
```

Comment:
No these are options of the `buildah` command, not buildah delete.

IE 

buildah -<tab> Shows them

buildah delete -<tab> shows these.